src/core/utils/misc.cpp

```diff
@@ -315,30 +317,57 @@ QString Utils::Misc::pythonVersionComplete() {
     return version;
 }
 
-// return best userfriendly storage unit (B, KiB, MiB, GiB, TiB)
+QString Utils::Misc::unitString(Utils::Misc::SizeUnit unit)
+{
+    return QCoreApplication::translate("misc",
+         units[static_cast<int>(unit)].source, units[static_cast<int>(unit)].comment);
+}
+
+// return best userfriendly storage unit (B, KiB, MiB, GiB, TiB, ...)
 // use Binary prefix standards from IEC 60027-2
 // see http://en.wikipedia.org/wiki/Kilobyte
 // value must be given in bytes
 // to send numbers instead of strings with suffixes
-QString Utils::Misc::friendlyUnit(qreal val, bool is_speed)
+bool Utils::Misc::friendlyUnit(qreal rawVal, qreal &val, Utils::Misc::SizeUnit &unit)
```

Comment:
>Did you mean quint64?

Yes. I made a typo. I mean **qulonglong**, which is the same.
>But does double has any meaning for this function at all?

We can't have a fractional number of bytes, but we can have fractional number of kibibytes, etc. So return type of **qreal** is correct.
>I guess there are negative sizes somewhere, and they are used to represent unknown size

This logic should be placed where it is required. Not here!